### PR TITLE
Fix UDT decoding: check both offset and field length NODEJS-175

### PR DESCRIPTION
**Pull request by jorgebay**

### CODE REVIEW

**Comment by cjihrig** [comment]
@jorgebay any update on when this will be merged?

**Comment by cjihrig** [comment]
When will a new release be published to npm?

File: test/integration/short/client-execute-prepared-tests.js [hunk 1]
```diff
@@ -575,6 +575,29 @@ describe('Client', function () {
...
+          client.connect.bind(client),
+          helper.toTask(client.execute, client, 'CREATE TYPE phone_change2 (alias text, number text, country_code int)'),
+          helper.toTask(client.execute, client, 'CREATE TABLE tbl_udt_change2 (id uuid PRIMARY KEY, phone_col2 frozen<phone_change2>)'),
+          function executeFewTimesFirst(next) {
+            var query = 'INSERT INTO tbl_udt_change2 (id, phone_col2) VALUES (?, ?)';
+            async.timesSeries(10, function (n, timesNext) {
+              client.execute(query, [types.Uuid.random(), { alias: n.toString(), number: n.toString()}], { prepare: true}, timesNext);
+            }, next)
+          },
+          helper.toTask(client.execute, client, 'ALTER TYPE phone_change2 ADD another text'),
```
Comment (cjihrig) [inline test/integration/short/client-execute-prepared-tests.js]: Can you add a test to verify that `another` is successfully inserted after the UDT is altered. I'm seeing it inserted as null.

Comment (jorgebay) [inline test/integration/short/client-execute-prepared-tests.js]: For the driver to consider the new field, you must re-prepare your statements. To force the statement to be prepared again, you must use `client.metadata.clearPrepared()`.

I've modified the test to insert new data in the new field:  fec079d7a3dd6fdfcd839bbbaa99ec66df53cdfc

**Pull request merged**